### DISCUSSION TITLE
Requiring that only one of two Options are required? (Union types are not supported)

### DISCUSSION THREAD

**Discussion by 1npo** [tiangolo/typer-cli#137] [Questions] [answered]
It seems that it's not possible to have CLI Options/Arguments that accept multiple types.

For example, creating a CLI Option with a Union[] type so that I can refer to some item either by its numerical ID or its string name:

```
from typing import Union
import typer

app = typer.Typer()

@app.command('item')
def item(item: Union[int, str] = typer.Option(None, '--item', '-i')):
	print(f'Doing a thing for item {item}')

def main():
	app()

if __name__ == '__main__':
	main()
```

Doesn't work. Running this yields this error: `AssertionError: Typer Currently doesn't support Union types`.

As a work-around, is it possible to make two CLI Options, and indicate that *only one of the two* Options are required?

For example, could you have an `--id` Option that accepts an integer, and a `--name` Option that accepts a string, and require that at least one of these two must be provided, but it can be either one?

It's not obvious in the documentation if this is possible (or I just haven't read it thoroughly enough).

Thanks!

**Comment by tiangolo** [comment, marked as answer]
This is probably a question for Typer, not for Typer-CLI. And now Typer includes all the logic and Typer-CLI command, so everything should go there.

Now, about your question, this functionality is provided by Click, so, as Click doesn't support more than one type, Typer can't do much about it. Maybe in the future I'll be able to optionally depend on Pydantic and extend the possibilities of what types are supported. But for now, it's not possible. :smiling_face_with_tear:
